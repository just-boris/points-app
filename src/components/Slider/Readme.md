## Basic example

```
<Slider title="Drag me!" value={0}></Slider>
```

## Example with some value

```
<Slider title="How much?" value={5}></Slider>
```
