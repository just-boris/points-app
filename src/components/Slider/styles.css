.Slider {
    display: block;
}

.Slider__title {
  min-width: 70px;
  text-transform: capitalize;
}

.Slider__control {
  width: 100%;
}
