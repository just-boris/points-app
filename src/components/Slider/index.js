import './styles.css';
import React from 'react';

export default function Slider({title, value}) {
  return (<label className="Slider row">
    <div className="row__col Slider__title">{title}</div>
    <div className="row__col row__col_flex">
      <input type="range" className="Slider__control" value={value} />
    </div>
    <div className="row__col">
      {value} pts
    </div>
  </label>);
}
