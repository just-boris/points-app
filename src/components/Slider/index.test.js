import React from 'react';
import {shallow} from 'enzyme';
import Slider from './';

describe('<Slider />', () => {
  function Page(component) {
    this.input = () => component.find('input');
  }

  function createComponent() {
    const onChange = jest.fn();
    const component = shallow(<Slider title="test" value={5} onChange={onChange} />)
    const page = new Page(component);
    return {component, page, onChange};
  }

  it('should render slider', () => {
    const {component} = createComponent();
    expect(component.get(0)).toMatchSnapshot();
  });

  it('should call handler when value was changed', () => {
    const {page, onChange} = createComponent();

    page.input().simulate('change', {target: {value: '3'}});

    expect(onChange).toBeCalledWith(3);
  })
});
