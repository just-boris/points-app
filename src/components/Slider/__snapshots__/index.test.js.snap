exports[`<Slider /> should render slider 1`] = `
<label
  className="Slider row">
  <div
    className="row__col Slider__title">
    test
  </div>
  <div
    className="row__col row__col_flex">
    <input
      className="Slider__control"
      max={10}
      onChange={[Function onChange]}
      type="range"
      value={5} />
  </div>
  <div
    className="row__col">
    5
     pts
  </div>
</label>
`;
